wandb/sdk/artifacts/artifacts_cache.py

```diff
@@ -35,8 +36,8 @@ def __init__(self, cache_dir: StrPath) -> None:
         mkdir_exists_ok(self._cache_dir)
         self._md5_obj_dir = os.path.join(self._cache_dir, "obj", "md5")
         self._etag_obj_dir = os.path.join(self._cache_dir, "obj", "etag")
-        self._artifacts_by_id: Dict[str, "ArtifactInterface"] = {}
-        self._artifacts_by_client_id: Dict[str, "LocalArtifact"] = {}
+        self._artifacts_by_id: Dict[str, "ArtifactInterface"] = CappedDict()
+        self._artifacts_by_client_id: Dict[str, "LocalArtifact"] = CappedDict()
```

Comment:
This breaks existing code: https://github.com/wandb/wandb/blob/918c698e2f9aca3890ba3c9ce2ba5829dd14710e/wandb/sdk/artifacts/storage_handlers/wb_local_artifact_handler.py#L56-L58